client/src/features/NetworkGraph/NetworkGraphContext.tsx

```diff
@@ -40,46 +19,12 @@ export function useNetworkGraph() {
 }
 
 export function NetworkGraphProvider({ children }: { children: ReactNode }) {
-    const initialData: NetworkGraphData = { nodes: [], links: [] };
-    const [data, setGraph] = useState<NetworkGraphData>(initialData);
-
-    useEffect(() => {
-        const stored = loadGraphFromStorage();
-        if (stored) {
-            setGraph(stored);
-        }
-    }, []);
-
-    const loadFromFiles = useCallback(async (filesLike : FileList | File[]) => {
-        try {
-            const files = Array.from(filesLike);
-
-            const allNodes: NetworkGraphData["nodes"] = [];
-            const allLinks: NetworkGraphData["links"] = [];
-
-            for (const file of files) {
-                if (!file.name.toLowerCase().endsWith(".json")) continue;
-
-                const text = await file.text();
-                const json = JSON.parse(text) as NetworkGraphData; // or a smaller "chunk" type
-
-                allNodes.push(...json.nodes);
-                allLinks.push(...json.links);
-            }
-
-            const graph: NetworkGraphData = { nodes: allNodes, links: allLinks };
+    const data  = useJsonFiles();
```

Comment:
This looks like a "useReadEverything()" to me since everything will be json now or in the future. See comment on issue tracker about breaking this down.